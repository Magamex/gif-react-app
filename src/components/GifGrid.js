import React, { Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const { data:images,loading } = useFetchGifs(category);



    // useEffect(() => {
    //     getGifs(category)
    //         .then(setImages)
    // }, [category]);

    return (
        <Fragment>
            <h3>{category}</h3>
            
            {loading && <p>Cargando...</p>}

            <div className='card-grid'> 
                     {
                         images.map(img =>(
                            <GifGridItem key={img.id} {...img}/> 
                         ))
                     }
            </div>

        </Fragment>
    )
}

GifGrid.prototype = {
    category: PropTypes.string.isRequired
}
