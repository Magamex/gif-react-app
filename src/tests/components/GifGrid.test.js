import React from 'react';
import '@testing-library/jest-dom'

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GifGrid', () => {
    test('Verificar que se genero correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const category= 'One Punch';
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/cualquiercosa.jpg',
                title: 'Cualquier cosa'
            }
        ]
        const category= 'One Punch';
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});