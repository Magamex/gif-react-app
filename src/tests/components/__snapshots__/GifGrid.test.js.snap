// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid Verificar que se genero correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
